text/0249-v2-experiments.md

```diff
@@ -199,6 +199,82 @@ Disadvantages:
+from the **public interface** of the stable release. The implementation
+(`.js` files) will be the same in both releases, so stable APIs in
+`aws-cdk-lib` can safely use experimental APIs internally (though not in
+their public API surfaces).
+
+**How does stripping APIs from the public work?**
+
+| TypeScript            | JavaScript | Java, Python, C#, Go            |
+|-----------------------|------------|---------------------------------|
+| Stripped from `.d.ts` | N/A        | No bindings generated by `jsii` |
+
+In TypeScript, the compiler will prevent usage of APIs that are not advertised
+in the `.d.ts` files. Users will be able to cast this away using `as any` though.
+They will also be able to confuse the compiler enough that they will get to pass
+struct properties that are technically `@experimental`.
+
+For JavaScript, an IDE will (hopefully) not autocomplete APIs missing from
+the `.d.ts` files, but otherwise nothing prevents users from calling the
+APIs, or passing struct properties that are technically experimental. On the
+other hand, JavaScript devs are by definition required to read the docs,
+so they can't miss the *experimental* banners displayed there.
+
+For jsii client languages, the bindings for experimental APIs will simply not
+exist so there's no way to work around that (though we'll have to check
+whether for example Python makes it possible to pass struct values that
+aren't in the declared API surface).
+
+**Vending**
+
+There are a number of different ways we can choose to distinguish the two builds.
+See Appendix D for an evaluation of all the possibilities.
+
+- Distinguish by package name
+  - Can't be used in **pip**: there's no way to have an app that uses the experimental
+    library work with a library that uses the stable library (no way to prevent the
+    transitive dependency package with a different name from being installed).
+- Distinguish by prerelease tag
+  - Can't be used in **Maven**: an app can't depend on stable versions only
+    (a range of `[1.60.0,2.0.0)` would also match `1.61.0-experimental`).
+- Distinguish by major version
+  - All package managers have a way to bring in a different version of a package,
+    although many of them will complain that declared dependencies aren't satisified.
+
+Distinguishing by major version is the only possibility that works across all
+package managers.
+
+Something like:
+
+| Scheme   | Stable   | Experimental            |
+|----------|----------|-------------------------|
+| Even/odd | `2.60.1` | `3.60.1`                |
+| +100     | `2.60.1` | `102.60.1`              |
+| <1       | `2.60.1` | `0.2.60-experimental.1` |
+
+**Summary**
+
+Advantages:
+
+1. Stable code can depend on unstable code.
+2. Stabilizing code will not be a breaking change for customers.
+3. Stabilizing code will be a simple operation for CDK developers.
+4. No additional management overhead of multiple packages/repos/etc.
```

Comment:
We're not interested in the repos argument here really. So I would just leave it as "No additional management overhead from splitting the code into multiple packages".